beacon-chain/sync/validate_blob.go

```diff
@@ -0,0 +1,193 @@
+	}
+	blob := sBlob.Message
+
+	// [REJECT] The sidecar is for the correct topic -- i.e. sidecar.index matches the topic {index}.
+	want := fmt.Sprintf("blob_sidecar_%d", blob.Index)
+	if !strings.Contains(*msg.Topic, want) {
+		log.WithFields(blobFields(blob)).Error("Sidecar blob does not match topic")
+		return pubsub.ValidationReject, fmt.Errorf("wrong topic name: %s", *msg.Topic)
+	}
+
+	// [IGNORE] The sidecar is not from a future slot (with a MAXIMUM_GOSSIP_CLOCK_DISPARITY allowance) --
+	// i.e. validate that sidecar.slot <= current_slot (a client MAY queue future blocks for processing at the appropriate slot).
+	genesisTime := uint64(s.cfg.chain.GenesisTime().Unix())
+	if err := slots.VerifyTime(genesisTime, blob.Slot, earlyBlockProcessingTolerance); err != nil {
+		return pubsub.ValidationIgnore, errors.Wrap(err, "blob too far into future")
+	}
+
+	// [IGNORE] The sidecar is from a slot greater than the latest finalized slot --
+	// i.e. validate that sidecar.slot > compute_start_slot_at_epoch(state.finalized_checkpoint.epoch)
+	startSlot, err := slots.EpochStart(s.cfg.chain.FinalizedCheckpt().Epoch)
+	if err != nil {
+		log.WithError(err).WithFields(blobFields(blob)).Error("Ignored block: could not calculate epoch start slot")
+		return pubsub.ValidationIgnore, err
+	}
+	if startSlot >= blob.Slot {
+		err := fmt.Errorf("finalized slot %d greater or equal to blob slot %d", startSlot, blob.Slot)
+		log.WithFields(blobFields(blob)).Warn(err)
+		return pubsub.ValidationIgnore, err
+	}
+
+	// [IGNORE] The blob's block's parent (defined by sidecar.block_parent_root) has been seen (via both gossip and non-gossip sources)
+	parentRoot := bytesutil.ToBytes32(blob.BlockParentRoot)
+	if !s.cfg.chain.HasBlock(ctx, parentRoot) {
+		if err := s.addPendingBlobToCache(sBlob); err != nil {
+			log.WithError(err).WithFields(blobFields(blob)).Error("Failed to add blob to queue")
+			return pubsub.ValidationIgnore, err
+		}
+		log.WithFields(blobFields(blob)).Warn("Ignored blob: parent block not found")
+		return pubsub.ValidationIgnore, nil
+	}
+
+	// [REJECT] The sidecar's block's parent (defined by sidecar.block_parent_root) passes validation.
+	blk, err := s.cfg.chain.GetBlock(ctx, parentRoot)
+	if err != nil {
+		log.WithError(err).WithFields(blobFields(blob)).Error("Failed to get parent block")
+		return pubsub.ValidationIgnore, err
+	}
+
+	// [REJECT] The sidecar is from a higher slot than the sidecar's block's parent (defined by sidecar.block_parent_root).
+	if blk.Block().Slot() >= blob.Slot {
+		err := fmt.Errorf("parent block slot %d greater or equal to blob slot %d", blk.Block().Slot(), blob.Slot)
+		log.WithFields(blobFields(blob)).Error(err)
+		return pubsub.ValidationReject, err
+	}
+
+	// [REJECT] The proposer signature, signed_blob_sidecar.signature,
+	// is valid with respect to the sidecar.proposer_index pubkey.
+	parentState, err := s.cfg.stateGen.StateByRoot(ctx, parentRoot)
+	if err != nil {
+		return pubsub.ValidationIgnore, err
```

Comment:
Shouldn't we reject here? After all we are unable to verify the signature.

Reply:
We are not able to get the parent state, which does not mean we cannot verify the signature. In this case, we probably shouldn't penalize the peer, but we do drop the object